hack/lib/protoc.sh

```diff
@@ -76,17 +91,36 @@ function kube::protoc::protoc() {
   )
 }
 
+# Generates $1/api.pb.go from the protobuf file $1/api.proto without using gogo
+# $1: Full path to the directory where the api.proto file is
+function kube::protoc::protoc() {
```

Comment:
we have a new pattern where functions not meant to be called directly have a `::internal::` segment (something thockin started I think), probably makes sense here if we're calling `kube::protoc::check_protoc` elsewhere but not here and making `kube::protoc::generate_proto_gogo` the "public api" ?

Reply:
like https://github.com/kubernetes/kubernetes/blob/e933a30523fe0d41b6769bae4662ecccd21bd4a1/hack/lib/golang.sh#L416

Reply:
I'm not sure if I understand that correctly. What we do here is trying to keep the same structure as existing but introducing a parallel implementation (gogo vs non goog).

Do you suggest that we need to refactor them into another location?